UsbIpServer/WslDistributions.cs

```diff
@@ -0,0 +1,83 @@
+﻿// SPDX-FileCopyrightText: Copyright (c) Microsoft Corporation
+//
+// SPDX-License-Identifier: GPL-2.0-only
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace UsbIpServer
+{
+    class WslDistributions
+    {
+        public record Distribution(string Name, IPAddress IPAddress);
+
+        readonly string? defaultDistro;
+
+        WslDistributions(List<Distribution> distributions, string? defaultDistro)
+        {
+            this.Distributions = distributions;
+            this.defaultDistro = defaultDistro;
+        }
+
+        public IReadOnlyCollection<Distribution> Distributions { get; }
+
+        public Distribution? DefaultDistribution => defaultDistro != null ? LookupByName(defaultDistro) : null;
+
+        public static async Task<WslDistributions> CreateAsync(CancellationToken cancellationToken)
+        {
+            var allDistrosResult = await ProcessUtils.RunCapturedProcessAsync("wsl.exe", new[] { "--list" }, Encoding.Unicode, cancellationToken);
+            if (allDistrosResult.ExitCode != 0)
+            {
+                throw new UnexpectedResultException($"WSL failed to list distributions: {allDistrosResult.StandardError}");
+            }
+
+            // Sample output:
+            //   Windows Subsystem for Linux Distributions:
+            //   Ubuntu(Default)
+            // We skip the first line, then look for a "(" to mark the default distro.
+            // This is similar to how Windows Terminal handles WSL.
+            // https://github.com/microsoft/terminal/blob/9e83655b0870f7964789a7a17ccfd232cab4945a/src/cascadia/TerminalSettingsModel/WslDistroGenerator.cpp#L120
+            var defaultDistro = allDistrosResult.StandardOutput.Split(Environment.NewLine, StringSplitOptions.RemoveEmptyEntries)
+                .Skip(1).FirstOrDefault(d => d.Contains('(', StringComparison.Ordinal));
+            if (defaultDistro != null)
+            {
+                var firstNonNameChar = defaultDistro.IndexOf('(', StringComparison.Ordinal);
+                if (firstNonNameChar != -1)
+                {
+                    defaultDistro = defaultDistro.Substring(0, firstNonNameChar).Trim();
+                }
+            }
+
+            var runningDistrosResult = await ProcessUtils.RunCapturedProcessAsync("wsl.exe", new[] { "--list", "--quiet", "--running" }, Encoding.Unicode, cancellationToken);
+            if (runningDistrosResult.ExitCode != 0)
```

Comment:
The compiler claims that this check and the similar exit code check below are always `false`. I'm pretty sure the compiler is wrong, but before I suppress the warning I like to have somebody else look at this to make sure I'm not missing anything.

Reply:
Confirmed. Just suppress.

Reply:
Suppressed and added a comment.